core/src/main/java/org/elasticsearch/index/Index.java

```diff
@@ -20,23 +20,38 @@
 package org.elasticsearch.index;
 
 import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
  *
  */
-public class Index implements Writeable<Index> {
+public class Index implements Writeable<Index>, ToXContent {
 
     public static final Index[] EMPTY_ARRAY = new Index[0];
+    private static final String INDEX_UUID_KEY = "index_uuid";
+    private static final String INDEX_NAME_KEY = "index_name";
+    public static final ObjectParser<Builder, Void> INDEX_PARSER = new ObjectParser<>("index", Index.Builder::new);
+    static {
+        INDEX_PARSER.declareString(Builder::name, new ParseField(INDEX_NAME_KEY));
+        INDEX_PARSER.declareString(Builder::uuid, new ParseField(INDEX_UUID_KEY));
+    }
 
     private final String name;
     private final String uuid;
 
     public Index(String name, String uuid) {
+        Objects.requireNonNull(name, "name is required");
```

Comment:
Let's leave the string constants out here, the stack trace is always enough to find out what went wrong and these string constants will just pollute the heap.